File: types/cluster.go
package types

// ClusterNextIdResponse maps to the GET /cluster/nextid API response
// from Proxmox. See:
// https://pve.proxmox.com/pve-docs/api-viewer/#/cluster/nextid
type ClusterNextIdResponse struct {
	VMID string `json:"data"`
}

// ClusterTasksResponse maps to the GET /cluster/tasks API response
// from Proxmox. See:https://pve.proxmox.com/pve-docs/api-viewer/#/nodes/{node}/lxc/{vmid}/clone
// https://pve.proxmox.com/pve-docs/api-viewer/#/cluster/tasks
type ClusterTasksResponse struct {
	Data []ClusterTasksData `json:"data"`
}

type ClusterTasksData struct {
	UPID string `json:"upid"`
}

File: pkg/typescript/src/types/cluster.ts
import { z } from 'zod';

export const ClusterNextIdResponseSchema = z.object({ VMID: z.string() });
export type ClusterNextIdResponse = z.infer<typeof ClusterNextIdResponseSchema>;

export const ClusterTasksDataSchema = z.object({ UPID: z.string() });
export type ClusterTasksData = z.infer<typeof ClusterTasksDataSchema>;

export const ClusterTasksResponseSchema = z.object({ Data: z.array(ClusterTasksDataSchema) });
export type ClusterTasksResponse = z.infer<typeof ClusterTasksResponseSchema>;
